app/controllers/application_controller/ci_processing.rb

```diff
@@ -1873,20 +1852,13 @@ def process_cloud_object_storage_buttons(pressed)
   end
 
   def cloud_object_store_button_operation(klass, task)
-    # Map to instance method name
-    case task
-    when "delete"
-      method = "#{task}_#{klass.name.underscore.to_sym}"
-      display_name = _(task.capitalize)
-    else
-      display_name = _(task.capitalize)
-      method = task = "#{klass.name.underscore.to_sym}_#{task}"
-    end
+    method = "#{task}_#{klass.name.underscore.to_sym}"
+    display_name = _(task.capitalize)
 
     items = []
 
     # Either a list or coming from a different controller
-    if @lastaction == "show_list" || %w(cloud_object_store_containers cloud_object_store_objects).include?(@display)
+    if @lastaction == "show_list" || !%w(cloud_object_store_container).include?(request.parameters["controller"])
```

Comment:
Instead of making these changes here in `ci_processing`, could you make all the `Physical Server` related changes in it's own controller, which would be `physical_server_controller`?